### PR TITLE
Support COLLECT_LIST in Flink and local processor

### PR DESCRIPTION
**Pull request by yunfengzhou-hub**
## What is the purpose of the change

This pull request adds support for COLLECT_LIST in FlinkProcessor and LocalProcessor

## Brief change log

- Support for COLLECT_LIST in FlinkProcessor and LocalProcessor on OverWindowTransform and SlidingWindowTransform.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: yes

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? docs

### CODE REVIEW

**Review by lindong28 — COMMENTED** [review]
Thanks for the PR! LGTM overall. Left one minor comment.

**Review by lindong28 — COMMENTED** [review]
Thanks for the PR! LGTM overall. Left one minor comment.

File: docs/content/concepts/window_aggregation.md [hunk 1]
```diff
@@ -78,5 +78,5 @@ FeatHub window transforms.
 | ROW_NUMBER   | Assigns a unique, sequential number to each row, starting with one, to the input rows within the window partition. |
 | COUNT        | Returns the number of input rows.                                                           |
 | VALUE_COUNTS | Returns a frequency map containing the counts of unique values of an expression or column across input rows. |
-
+| COLLECT_LIST | Returns an array containing all values in an expression or column across input rows.        |
```
Comment (lindong28) [inline docs/content/concepts/window_aggregation.md:81]: The aggregation function should not need to know expression/column.

How about the following doc:

COLLECT_LIST: returns a list that contains the ordered list of input values.

I will create a separate PR to improve doc for other functions.

**Comment by yunfengzhou-hub** [comment]
Thanks for the comments @lindong28. I have updated the PR according to the comments.

**Comment by lindong28** [comment]
Thanks for the update. LGTM.

**Pull request merged**